File: apps/api/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    notification_type: str
    title: str
    message: str
    related_chore_id: int | None
    related_household_id: int | None
    is_read: bool
    created_at: datetime
    read_at: datetime | None


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    unread_count: int


class NotificationActionResponse(BaseModel):
    updated_count: int


class NotificationDeleteResponse(BaseModel):
    deleted_count: int

File: apps/web/src/types/notification.ts
export type NotificationItem = {
  id: number;
  notification_type: string;
  title: string;
  message: string;
  related_chore_id: number | null;
  related_household_id: number | null;
  is_read: boolean;
  created_at: string;
  read_at: string | null;
};

export type NotificationListResponse = {
  items: NotificationItem[];
  unread_count: number;
};

export type NotificationActionResponse = {
  updated_count: number;
};

export type NotificationDeleteResponse = {
  deleted_count: number;
};
